### ISSUE TITLE
Windows: improper HiDPI support in multi-DPI setups

### ISSUE THREAD

**Issue by JW-AV** [gaphor/gaphor#1558]
(I recognize this is probably not Gaphor's fault but misbehavior of GTK/GDK. Maybe there is something one could do here though?)

On Windows (11) with multiple Displays connected, only one of which is HiDPI (4K), starting Gaphor on that HiDPI-display makes all UI-elements appear unusably small.

Setting the environment variable `GDK_SCALE=2` makes things way better (unfortunately, fractional scaling like `GDK_SCALE=1.5` doesn't work). On the other hand, `GDK_WIN32_DISABLE_HIDPI=1` makes the UI scale correctly but very blurry to the point of eye pain, rendering it unusable, as well.

### OS
- [X] Windows (11)

### Version
Version of Gaphor: 2.10.0

**Comment by danyeaw** [comment]
Hi @JW-AV thanks for the report!

I don't have a HiDPI display to test this with. Is it still a problem if you only have the HiDPI display connected?

It looks like [some support](https://bugzilla.gnome.org/show_bug.cgi?id=768081) was added in GTK 3.22 and GTK4 by Chun-wei Fan who makes a lot of the GTK contributions on the Windows side. However you might have run in to a bug in the implementation.

**Comment by amolenaar** [comment]
Indeed, this is more of a GTK issue. There is not a lot we can do in a lot about in Gaphor itself. I would expect it to be better with GTK4.

**Comment by danyeaw** [comment]
If you are interested in seeing if GTK4 does fix the issue, you could follow the instructions to setup a development environment for Windows using gvsbuild at https://gaphot.readthedocs.io, but instead of gtk3 and gtksourceview build with gtk4 and gtksourceview5.

If you get all that working then you can launch gaphor with `GAPHOR_USE_GTK4 gaphor`.
